### ISSUE TITLE
Keymaps should record only 1 entry to the modification history

### ISSUE THREAD

**Issue by k2d222** [helix-editor/helix#1504]
Custom *sequence keymaps*  (defined in user's `config.toml`) will execute the commands in sequence but also record each modification to the history. Undoing will undo the individual components of the *sequence keymap*, not the keymap as a whole.

**An example**:
`A-s = [ "extend_to_line_bounds", "delete_selection", "paste_after" ]`
This keymap will swap a line with the one below. But it requires 2 undos for `paste_after` and `delete_selection`

**Possible solution**:
Add commands to start / stop recording individual modifications to history. When recording is stopped, modifications will update the last history `Revision` instead of adding new ones.

If this feature request is found interesting, I would happily have a go at implementing it :smiley:

**Comment by Omnikar** [comment]
Yeah, it was decided to momentarily skip this behaviour in the original implementation: https://github.com/helix-editor/helix/pull/589#issuecomment-909356419

I do agree though that it is annoying; I can go back and implement it, and it'll be useful for macros as well, since they have the same issue.

**Comment by Omnikar** [comment]
So, this turns out to be a bit tricky due to the fact that documents keep their own individual histories, and command sequences are capable of switching the current document.
